de1plus/profile.tcl

```diff
@@ -513,5 +519,282 @@ namespace eval ::profile {
+    proc read_legacy { profile_path {extra_vars {}} } {
+        if { [file pathtype $profile_path] eq "relative" && [file dirname $profile_path] eq "." } {
+            set profile_path "[homedir]/profiles/[file tail $profile_path]"
+        }
+        if { [file extension $profile_path] eq "" } {
+            append profile_path ".tcl"
+        }
+        if { ![file exists $profile_path] } {
+            msg -WARNING [namespace current] load_profile: "profile file '$profile_path' not found"
+            return
+        }
+        
+        msg -INFO [namespace current] read_profile: "reading file '$profile_path'"
+        array set profile {}
+    
+        # Set defaults in case they are not defined in the profile file 
+        set profile(final_desired_shot_volume_advanced_count_start) 0
+        set profile(settings_profile_type) "settings_2a"
+        set profile(beverage_type) "espresso"
+    
+        try {
+            array set arrfile [encoding convertfrom utf-8 [read_binary_file $profile_path]]
+        } on error err {
+            msg -WARNING [namespace current] read_profile: "error reading profile '$fn': $err"
+            return
+        }
+    
+        foreach k [concat [::profile_vars] $extra_vars] {
+            if { [info exists arrfile($k)] } {
+                set profile($k) $arrfile($k)
+            } elseif { ![info exists profile($k)] } {
+                set profile($k) {}
+            }
+        }
+        
+        set profile(settings_profile_type) [fix_profile_type $profile(settings_profile_type)]
+        
+        if { $profile(settings_profile_type) eq "settings_2c" && $profile(final_desired_shot_weight) ne "" &&
+                $profile(final_desired_shot_weight_advanced) eq "" } {
+            msg -NOTICE [namespace current] load_profile: "Using a default for final_desired_shot_weight_advanced " \
+                "from final_desired_shot_weight of $profile(final_desired_shot_weight)"
+            set profile(final_desired_shot_weight_advanced) $profile(final_desired_shot_weight)
+        }
+    
+        # pre-set the shot volume, to the shot weight, if importing an old shot definition that doesn't have a an end volume 
+        if { $profile(final_desired_shot_volume) eq "" } {
+            msg -NOTICE [namespace current] load_profile: "pre-set the shot volume, to the shot weight, if importing an old shot definition" \
+                "that doesn't have a an end volume"
+            set profile(final_desired_shot_volume) $profile(final_desired_shot_weight)
+        }
+    
+        if { $profile(advanced_shot) eq {} } {
+            # Ensure the profile's advanced_shot variable is always defined
+            switch $profile(settings_profile_type) \
+                settings_2a {
+                    array set temp_profile [pressure_to_advanced_list profile]
+                    set profile(advanced_shot) $temp_profile(advanced_shot)
+                } settings_2b {
```

Comment:
I dont think they need to be in the same line like if / else

Reply:
Do you mean that the `switch` line would work without the final "\"?

Reply:
settings_2a {
    } 
   settings_2b {

Reply:
Ah, ok, sure can be changed, do it on merge if you want